lib/cable_ready/channel.rb

```diff
@@ -2,245 +2,35 @@
-    #   }, ...],
-    #
-    #   text_content: [{
-    #     selector: "string",
-    #     text:     "string"
-    #   }, ...]
-    #
-    #   insert_adjacent_html: [{
-    #     selector:      "string",
-    #     focus_selector: "string",
-    #     position:      "string",
-    #     html:          "string"
-    #   }, ...],
-    #
-    #   insert_adjacent_text: [{
-    #     selector: "string",
-    #     position: "string",
-    #     text:     "string"
-    #   }, ...],
-    #
-    #   remove: [{
-    #     selector:      "string",
-    #     focus_selector: "string,
-    #   }, ...],
-    #
-    #   set_property: [{
-    #     name:     "string",
-    #     selector: "string",
-    #     value:    "string"
-    #   }, ...],
-    #
-    #   set_value: [{
-    #     selector: "string",
-    #     value:    "string"
-    #   }, ...],
-    #
-    #   # Attribute Mutations .........................................................................................
-    #
-    #   set_attribute: [{
-    #     selector: "string",
-    #     name:     "string",
-    #     value:    "string"
-    #   }, ...],
-    #
-    #   remove_attribute: [{
-    #     selector: "string",
-    #     name:     "string"
-    #   }, ...],
-    #
-    #   # CSS Class Mutations .........................................................................................
-    #
-    #   add_css_class: [{
-    #     selector: "string",
-    #     name:     "string"
-    #   }, ...],
-    #
-    #   remove_css_class: [{
-    #     selector: "string",
-    #     name:     "string"
-    #   }, ...],
-    #
-    #   # Style Mutations ...........................................................................................
-    #
-    #   set_style: [{
-    #     selector: "string",
-    #     name:     "string",
-    #     value:    "string"
-    #   }, ...],
-    #
-    #   # Dataset Mutations ...........................................................................................
-    #
-    #   set_dataset_property: [{
-    #     selector: "string",
-    #     name:     "string",
-    #     value:    "string"
-    #   }, ...],
-    # }
-    def initialize(identifier)
+    def initialize(identifier, available_operations)
       @identifier = identifier
-      @operations = stub
-    end
-
-    def reset
-      @operations = stub
+      @available_operations = available_operations
+      @operations = Hash.new { |hash, operation| hash[operation] = [] }
+      available_operations.each do |available_operation, implementation|
+        define_singleton_method available_operation, &implementation
+      end
     end
 
     def broadcast(clear)
       operations.select! { |_, list| list.present? }
       operations.deep_transform_keys! { |key| key.to_s.camelize(:lower) }
       ActionCable.server.broadcast identifier, "cableReady" => true, "operations" => operations
-      reset if clear
+      @operations = Hash.new { |hash, operation| hash[operation] = [] } if clear
```

Comment:
Maybe extract this in its own private method `reset_operations` or `clear_operations`